### PR TITLE
Feature/edit group

### PR DESCRIPTION
**Pull request by VladimirTomko**
## Summary

Functionality to edit groups
Improved styling & responsivity
Added more color options
Fixed bugs
Minor changes

## Screenshot(s)

n/a

## Type of change

- [x] Bug fix (a change which fixes an issue)
- [x] Enhancement (a change which improves existing functionality)
- [x] New feature (a change which adds new functionality)
- [x] Chore (non-critical update intended to improve the codebase)

### CODE REVIEW

File: src/pages/Board/EditGroupForm.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  PopoverHeader,
+  PopoverBody,
+} from '@chakra-ui/react'
+import { EditIcon } from '@chakra-ui/icons'
+
+export const EditGroupForm = ({
+  boardColor,
+  currentGroupName,
+  currentGroupId,
+  handleEditNewGroup,
```
Comment (misindelka) [inline src/pages/Board/EditGroupForm.jsx:20]: here to handleEditGroup

File: src/pages/Board/EditGroupForm.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}) => {
+  const [editedGroupName, setEditedGroupName] = React.useState(currentGroupName)
+
+  //   React.useEffect(() => {
```
Comment (misindelka) [inline src/pages/Board/EditGroupForm.jsx:25]: remove these comments

File: src/pages/Board/EditGroupForm.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          <Input
+            name="name"
+            color="black"
+            variant="flushed"
+            placeholder="Group name here"
+            value={editedGroupName}
+            onChange={(e) => setEditedGroupName(e.target.value)}
+          />
+          <Button
+            isDisabled={!editedGroupName}
```
Comment (misindelka) [inline src/pages/Board/EditGroupForm.jsx:70]: please clarify it to me , thx

File: src/pages/Board/Board.jsx [hunk 4]
```diff
@@ -27,6 +27,10 @@ const Board = () => {
     createTaskGroup(newGroup.boardId, newGroup.name)
   }
 
+  const handleEditTaskGroup = (taskGroupId) => {
+    updateTaskGroup(taskGroupId) // data
```
Comment (misindelka) [inline src/pages/Board/Board.jsx:31]: updating just id or ??

**Review by misindelka — APPROVED** [review]
nice just few things need to be fixed and some explained

File: src/pages/components/TaskGroups.jsx [hunk 5]
```diff
@@ -65,6 +73,10 @@ export const TaskGroups = ({ boardId, handleCreateGroup, boardColor }) => {
     updateTask(currentTask.id, data)
   }
 
+  const handleEditNewGroup = (taskGroupId, data) => {
```
[the remark this answers is not in the document]
Comment (VladimirTomko) [reply src/pages/components/TaskGroups.jsx:77]: fixed

File: src/pages/components/TaskGroups.jsx [hunk 6]
```diff
@@ -91,7 +103,12 @@ export const TaskGroups = ({ boardId, handleCreateGroup, boardColor }) => {
             p="5"
             fontSize="2xl"
           >
-            <EditIcon float="left" w="4" h="8" />
+            <EditGroupForm
+              currentGroupId={group.id}
+              handleEditNewGroup={handleEditNewGroup}
```
[the remark this answers is not in the document]
Comment (VladimirTomko) [reply src/pages/components/TaskGroups.jsx:111]: fixed

File: src/pages/Board/EditGroupForm.jsx (hunk 1 above)
Comment (VladimirTomko) [reply src/pages/Board/EditGroupForm.jsx:20]: fixed

File: src/pages/Board/EditGroupForm.jsx (hunk 2 above)
Comment (VladimirTomko) [reply src/pages/Board/EditGroupForm.jsx:25]: fixed

File: src/pages/Board/EditGroupForm.jsx (hunk 3 above)
Comment (VladimirTomko) [reply src/pages/Board/EditGroupForm.jsx:70]: fixed

File: src/pages/Board/Board.jsx (hunk 4 above)
Comment (VladimirTomko) [reply src/pages/Board/Board.jsx:31]: fixed

**Review by misindelka — APPROVED** [review]
fixed

**Pull request merged**